components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/asyncapi/models/AsyncApiV2Parser.java

```diff
@@ -0,0 +1,153 @@
+import org.wso2.carbon.apimgt.spec.parser.definitions.BaseAsyncApiV2Parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * New AsyncAPI v2 parser used to process AsyncAPI 2.x specifications.
+ * This class extends BaseAsyncApiV2Parser and
+ * provides version-specific  parsing capabilities for AsyncAPI 2.x definitions.
+ */
+public class AsyncApiV2Parser extends BaseAsyncApiV2Parser {
+
+    private static final Log log = LogFactory.getLog(AsyncApiV2Parser.class);
+
+    /**
+     * Extracts and returns the set of scopes defined in the given AsyncAPI configuration.
+     *
+     * @param resourceConfigsJSON the AsyncAPI configuration in JSON format
+     * @return a set of Scope bjects defined in the configuration
+     * @throws APIManagementException if an error occurs while extracting the scopes
+     */
+    @Override
+    public Set<Scope> getScopes(String resourceConfigsJSON) throws APIManagementException {
+        Set<Scope> scopeSet = AsyncApiParserUtil.getScopesFromAsyncAPIConfig(resourceConfigsJSON);
+        return scopeSet;
+    }
+
+    /**
+     * Validates the given AsyncAPI definition against the AsyncAPI JSON HyperSchema.
+     *
+     * @param apiDefinition      the AsyncAPI definition to be validated, in JSON format
+     * @param returnJsonContent  whether the validated content should be returned in JSON format
+     * @return an APIDefinitionValidationResponse containing validation results and messages
+     * @throws APIManagementException if an error occurs during validation or schema processing
+     */
+    @Override
+    public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinition, boolean returnJsonContent)
+            throws APIManagementException {
+
+        APIDefinitionValidationResponse validationResponse = new APIDefinitionValidationResponse();
+        String protocol = StringUtils.EMPTY;
+        boolean validationSuccess = false;
+        List<String> validationErrorMessages = new ArrayList<>();
+
+        try {
+            validationSuccess = AsyncApiParserUtil.validateAsyncApiContent(apiDefinition, validationErrorMessages);
+        } catch (Exception e) {
+            // unexpected problems during validation/parsing
+            String msg = "Error occurred while validating AsyncAPI definition: " + e.getMessage();
+            log.error(msg, e);
+            throw new APIManagementException(msg, e, ExceptionCodes.ERROR_READING_ASYNCAPI_SPECIFICATION);
+        }
+
+        if (validationSuccess) {
+            AsyncApiDocument asyncApiDocument = (AsyncApiDocument) Library.readDocumentFromJSONString(apiDefinition);
+            ArrayList<String> endpoints = new ArrayList<>();
+            AsyncApiServers servers = asyncApiDocument.getServers();
+            if (servers != null && servers.getItems() != null && !servers.getItems().isEmpty() &&
+                    servers.getItems().size() == 1)
```

Comment:
what happens if servers has multiple elements? shouldnt we always get the first one without checking if it only has one